tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -2887,8 +2887,13 @@ std::tuple<llvm::Value*, llvm::Value*> CalculateYXCoordinateWithinTile(
     int64 threads_per_tile) {
   // Calculate the starting element coordinate within a tile for the current
   // thread, (y, x) from thread_id.
-  llvm::Value* thread_id = llvm_ir::EmitCallToIntrinsic(
-      llvm::Intrinsic::nvvm_read_ptx_sreg_tid_x, {}, {}, builder);
+#ifdef GOOGLE_CUDA
```

Comment:
@sumesh13 I'd recommend changing logic here to:

```
#if TENSORFLOW_USE_ROCM
// amdgpu-specific logic
#else
// nvptx-specific logic
#endif
```

to my understanding `GOOGLE_CUDA` may *NOT* be defined in when XLA is enabled in certain cases.